### DISCUSSION TITLE
White page when adding domain to Github Pages

### DISCUSSION THREAD

**Discussion by byloupom** [community/community#23066] [Other Feature Feedback, Questions, & Ideas]
<p>Hi community!</p>
<p>For my upcoming wedding, I’ve created a react app in my github repo that works fine on my Github Pages repo <a href="https://github.com/byloupom/l-et-m" class="inline-onebox" rel="noopener nofollow ugc">GitHub - byloupom/l-et-m</a> when there is no custom domain :<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/c/d/cd6a4fe168c89667dafbe3a6db09b05701bd05ba.png" data-download-href="/uploads/short-url/tjbExG6qV4uDVzxCvKkwpt6dWSC.png?dl=1" title="Screen" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/109140212/181074652-c62ad6aa-e444-40f7-a44a-aa34e329ada5.png" alt="Screen" data-base62-sha1="tjbExG6qV4uDVzxCvKkwpt6dWSC" width="690" height="407" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/d/cd6a4fe168c89667dafbe3a6db09b05701bd05ba_2_690x407.png, https://aws1.discourse-cdn.com/github/optimized/3X/c/d/cd6a4fe168c89667dafbe3a6db09b05701bd05ba_2_1035x610.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/c/d/cd6a4fe168c89667dafbe3a6db09b05701bd05ba_2_1380x814.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/d/cd6a4fe168c89667dafbe3a6db09b05701bd05ba_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">Screen</span><span class="informations">1692×1000 163 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>But when I connect my custom domain <a href="https://louis-et-marianne.com/" rel="noopener nofollow ugc">https://louis-et-marianne.com/</a> following all instructions to the letter, I get a white blank page.</p>
<p>The console is full of 404 errors while attempting at getting the main.js, main.css, <a href="http://favicon.io" rel="noopener nofollow ugc">favicon.io</a>, and manifest.json files.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/7/9/794becd256e6ca039693c2d7d5062c81b6505be2.png" data-download-href="/uploads/short-url/hj2rrC6PFwryIP6QCIWrzLiCy0a.png?dl=1" title="Console" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/109140212/181074653-7e23f403-a6a6-4c97-b56d-48bf1b0bb798.png" alt="Console" data-base62-sha1="hj2rrC6PFwryIP6QCIWrzLiCy0a" width="690" height="183" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/7/9/794becd256e6ca039693c2d7d5062c81b6505be2_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">Console</span><span class="informations">1141×304 18.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div> - the app title displays fine in my browser tab, but it’s a blank page nonetheless.
<p>Here is my Google Domain setup :<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/4/3/4312137b75c701f3a50cb8c2bf822ab902472f8a.png" data-download-href="/uploads/short-url/9zkIxuLcJ88tTEtz6Y0Z4OwmvXQ.png?dl=1" title="GoogleDomains" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/109140212/181074654-d6e91a7c-f799-4172-b53f-af0b523ba16f.png" alt="GoogleDomains" data-base62-sha1="9zkIxuLcJ88tTEtz6Y0Z4OwmvXQ" width="690" height="191" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/4/3/4312137b75c701f3a50cb8c2bf822ab902472f8a_2_690x191.png, https://aws1.discourse-cdn.com/github/optimized/3X/4/3/4312137b75c701f3a50cb8c2bf822ab902472f8a_2_1035x286.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/4/3/4312137b75c701f3a50cb8c2bf822ab902472f8a_2_1380x382.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/4/3/4312137b75c701f3a50cb8c2bf822ab902472f8a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">GoogleDomains</span><span class="informations">1542×427 27.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>Here is Github setup:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/7/a/7a53dc4d831a6f35f0002ea775a7fd7a6c6f0e1c.png" data-download-href="/uploads/short-url/hs9V9roKc00gvaRs9hT5rNNhNnu.png?dl=1" title="githubsetup" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/109140212/181074655-ae7687ae-587d-43ba-b518-0ac27a2efab9.png" alt="githubsetup" data-base62-sha1="hs9V9roKc00gvaRs9hT5rNNhNnu" width="690" height="384" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/7/a/7a53dc4d831a6f35f0002ea775a7fd7a6c6f0e1c_2_690x384.png, https://aws1.discourse-cdn.com/github/optimized/3X/7/a/7a53dc4d831a6f35f0002ea775a7fd7a6c6f0e1c_2_1035x576.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/7/a/7a53dc4d831a6f35f0002ea775a7fd7a6c6f0e1c_2_1380x768.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/7/a/7a53dc4d831a6f35f0002ea775a7fd7a6c6f0e1c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">githubsetup</span><span class="informations">1630×909 87.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>In my local package.json I’ve tried several values for homepage but reverted for now to my Github Repo :<br>
<code>"homepage": "https://byloupom.github.io/l-et-m/",</code></p>
<p>Any help would be much appreciated! (Note: I’m a beginner at coding!)</p>
<p>Cheers,<br>
Louis</p>

**Comment by jfullstackdev** [comment]
<p>I can access the icon, meaning it’s already hosted on GitHub Pages from <code>gh-pages</code> branch</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/2/d/2d4fac57052d553c5915986a7226b0be1539d662.png" data-download-href="/uploads/short-url/6sQbtZ6XFonhgBo8Oevd5617562.png?dl=1" title="image"><img src="https://user-images.githubusercontent.com/47092464/181074657-916bddf7-36c0-4b8c-9a88-cfba0a5c04ba.png" alt="image" data-base62-sha1="6sQbtZ6XFonhgBo8Oevd5617562" width="690" height="373" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/2/d/2d4fac57052d553c5915986a7226b0be1539d662_2_690x373.png, https://aws1.discourse-cdn.com/github/optimized/3X/2/d/2d4fac57052d553c5915986a7226b0be1539d662_2_1035x559.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/2/d/2d4fac57052d553c5915986a7226b0be1539d662_2_1380x746.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/2/d/2d4fac57052d553c5915986a7226b0be1539d662_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1920×1040 91 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>but there is no such thing as</p>
<p><code>src="/l-et-m/static/js/main.f898d1a7.js"</code> because you are at the <code>gh-pages</code>, it will work locally but not on GitHub Pages</p>
<p>try the absolute URL but not very sure of that too</p>
<p><a href="https://louis-et-marianne.com/static/js/main.f898d1a7.js">https://louis-et-marianne.com/static/js/main.f898d1a7.js</a></p>

**Comment by byloupom** [comment, marked as answer]
<p>Thanks!</p>
<p>I’ve removed the repository name in package.json from</p>
<pre><code class="lang-auto">"homepage": "https://byloupom.github.io/l-et-m/",
</code></pre>
<p>to</p>
<pre><code class="lang-auto">"homepage": "https://byloupom.github.io/",
</code></pre>
<p>And it worked! What is misleading is that whenever you deploy to Github Pages it removes the custom domain from the repository settings, and therefore you think there is a bug, but there is not.</p>
<p>There are plenty ressources online about how to not reset Github Pages custom domain on deploy.</p>
<p>Many thanks!</p>

**Reply by Welly0007** [reply]
also check if you are using http or https, as when https was enforced using github pages, there was still a white page,
also thanks for your solution, it is a life saver

**Reply by Wissem7155hack** [reply]
Hi, 
after deploying my website using custom domain it shows a white blank screen why?!!! everything is working with no errors but just no content i need help asap please
thanks

**Comment by jashwinsingh11** [comment]
I faced the same issue when deploying a React app to GitHub Pages with a custom domain, where the site worked perfectly on the default GitHub Pages URL but showed a blank white page with multiple 404 errors after connecting the custom domain. The root cause was an incorrect homepage value in package.json. Since the app was initially hosted under a repository subpath, the homepage was set to the GitHub Pages repo URL, which caused React to generate asset paths that included the repo name. When switching to a custom domain, the app is served from the root of the domain, so those paths became invalid and resulted in missing JavaScript and CSS files. I fixed this by updating the homepage field to point to the custom domain root, rebuilding the app, and redeploying it to GitHub Pages. After doing this and ensuring the custom domain was correctly configured in the GitHub Pages settings, the 404 errors disappeared and the application loaded normally without the blank page.

**Comment by PabloSandov169o** [comment]
quien lo diria
